On Solaris, it's not possible to list DTrace probes with readelf.

Should be reported upstream.

--- Python-3.15.0/Lib/test/test_dtrace.py
+++ Python-3.15.0/Lib/test/test_dtrace.py
@@ -515,6 +515,9 @@ class BPFTraceOptimizedTests(TraceTests,
 class CheckDtraceProbes(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
+        if sys.platform.startswith("sunos"):
+            raise unittest.SkipTest("DTrace probes are not visible to readelf on Solaris.")
+
         readelf_major_version, readelf_minor_version = cls.get_readelf_version()
         if support.verbose:
             print(f"readelf version: {readelf_major_version}.{readelf_minor_version}")
